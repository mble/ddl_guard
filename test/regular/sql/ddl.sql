SELECT current_setting('ddl_guard.enabled', true) = 'on' AS ddl_guard_enabled;
SELECT current_setting('is_superuser', true) = 'on' AS is_superuser;
CREATE TABLE IF NOT EXISTS foobar (id serial PRIMARY KEY);
CREATE OR REPLACE FUNCTION foobar_trigger() RETURNS TRIGGER AS $$
BEGIN
    RETURN NEW;
END;
$$ LANGUAGE plpgsql;
DROP EXTENSION ddl_guard;
SET ddl_guard.enabled TO off;
