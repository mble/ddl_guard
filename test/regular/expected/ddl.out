SELECT current_setting('ddl_guard.enabled', true) = 'on' AS ddl_guard_enabled;
 ddl_guard_enabled 
-------------------
 t
(1 row)

SELECT current_setting('is_superuser', true) = 'on' AS is_superuser;
 is_superuser 
--------------
 f
(1 row)

CREATE TABLE IF NOT EXISTS foobar (id serial PRIMARY KEY);
WARNING:  ddl_guard: ddl detected, sentinel file written
CREATE OR REPLACE FUNCTION foobar_trigger() RETURNS TRIGGER AS $$
BEGIN
    RETURN NEW;
END;
$$ LANGUAGE plpgsql;
WARNING:  ddl_guard: ddl detected, sentinel file written
DROP TABLE IF EXISTS foobar;
WARNING:  ddl_guard: ddl detected, sentinel file written
DROP FUNCTION IF EXISTS foobar_trigger;
WARNING:  ddl_guard: ddl detected, sentinel file written
